tools/generate-meta.ts

```diff
@@ -14,121 +14,16 @@
-const METADATA_FOLDER = join(
-  toUnixPath(import.meta.dirname),
-  '../sonar-plugin/javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/',
-);
-const sonarWayProfileFile = join(METADATA_FOLDER, `Sonar_way_profile.json`);
-const sonarWayProfile = JSON.parse(readFileSync(sonarWayProfileFile, 'utf-8'));
-
-// Check rule rspec metadata docs in https://github.com/SonarSource/rspec/blob/master/docs/metadata.adoc
-async function generateMetaForRule(ruleDir: string, ruleId: string) {
-  const ruleRspecMeta = JSON.parse(
-    readFileSync(join(METADATA_FOLDER, `${ruleId}.json`), 'utf-8'),
-  ) as rspecMeta;
-  if (!typeMatrix[ruleRspecMeta.type]) {
-    console.log(`Type not found for rule ${ruleId}`);
-  }
-  const metadata: Omit<RuleMetaData<any, any>, 'schema' | 'messages'> = {
-    type: typeMatrix[ruleRspecMeta.type],
-    docs: {
-      description: ruleRspecMeta.title,
-      recommended: sonarWayProfile.ruleKeys.includes(ruleId),
-      //url: `https://github.com/SonarSource/rspec/blob/master/rules/${ruleId}/javascript/rule.adoc`,
-      url: `https://sonarsource.github.io/rspec/#/rspec/${ruleId}/javascript`,
-      requiresTypeChecking: ruleRspecMeta.tags.includes('type-dependent'),
-    },
-  };
-  if (ruleRspecMeta.quickfix === 'covered') {
-    metadata.fixable = 'code';
-  }
-  if (ruleRspecMeta.status === 'deprecated') {
-    metadata.deprecated = true;
-  }
-
-  let schema = false;
-  try {
-    schema = JSON.parse(
-      readFileSync(join(METADATA_FOLDER, 'schemas', `${ruleId}-schema.json`), 'utf-8'),
-    );
-  } catch {}
-  const metaFile = join(ruleDir, ruleId, 'generated-meta.ts');
-  writeFileSync(
-    metaFile,
-    await prettier.format(
-      `${header}
-
-// DO NOT EDIT! This file is autogenerated by "npm run generate-meta"
-
-export const meta = ${JSON.stringify(metadata, null, 2)};
-
-export const sonarKey = '${ruleId}';
-${
-  schema
-    ? `import { JSONSchema4 } from '@typescript-eslint/utils/json-schema'; 
-export const schema = ${JSON.stringify(schema, null, 2)} as const satisfies JSONSchema4;`
-    : ''
-}
-`,
-      { ...(prettierOpts as prettier.Options), filepath: metaFile },
-    ),
-  );
-}
-
-async function generateMetaForRules(ruleDir: string) {
-  const files = readdirSync(ruleDir, { withFileTypes: true });
-  for (const file of files) {
-    if (!ruleRegex.test(file.name)) {
-      continue;
-    }
-    if (file.isDirectory()) {
-      await generateMetaForRule(ruleDir, file.name);
-    }
-  }
+/**
+ * Generate packages/jsts/src/rules/SXXXX/generated-meta.ts on each rule
+ * with data coming from the RSPEC json files. This data fills in the Rule ESLint metadata
+ * as well as the JSON schema files available in
+ * "sonar-plugin/javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/schemas"
+ */
+for (const file of await listRulesDir()) {
+  await generateMetaForRule(file);
```

Comment:
not sure what the proper art is, but usually you shouldn't have await in a for loop. Better to hand it off to the language to manage these promises. As in:
```suggestion
Promise.all((await listRulesDir()).map( async (file) => await generateMetaForRule(file)));
```

Come to think of it, we didn't have many rules/lints in PHP/Hack, but not having await's in loops was one of them. Usually as you were looping though friends lists or photos to retrieve some data, as its a slow operation (DB reads), having it in an abstract map was one of the rules.

Reply:
I don't agree here... i prefer to do sequential to not launch 1k+ read IO at the same time.